components/dashboard/src/repositories/detail/workspaces/WorkpaceSizeOptions.tsx

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import type { Configuration } from "@gitpod/public-api/lib/gitpod/v1/configuration_pb";
+import React, { useCallback, useState } from "react";
+import { useWorkspaceClasses } from "../../../data/workspaces/workspace-classes-query";
+import { Label } from "@podkit/forms/Label";
+import { RadioGroup, RadioGroupItem } from "@podkit/forms/RadioListField";
+import { TextMuted } from "@podkit/typography/TextMuted";
+import { Heading2 } from "@podkit/typography/Headings";
+import { ConfigurationSettingsField } from "../ConfigurationSettingsField";
+import { useUpdateProject } from "../../../data/projects/project-queries";
+import { useToast } from "../../../components/toasts/Toasts";
+import { LoadingButton } from "@podkit/buttons/LoadingButton";
+
+interface Props {
+    configuration: Configuration;
+}
+
+export const ConfigurationWorkspaceSizeOptions = ({ configuration }: Props) => {
+    const [selectedValue, setSelectedValue] = useState(
+        configuration.workspaceSettings?.workspaceClass || "g1-standard",
+    );
+    const classChanged = selectedValue !== configuration.workspaceSettings?.workspaceClass;
+
+    const updateProject = useUpdateProject();
+    const { data: classes, isError, isLoading } = useWorkspaceClasses();
+
+    const { toast } = useToast();
+
+    const setWorkspaceClass = useCallback(
+        async (e: React.FormEvent) => {
+            e.preventDefault();
+
+            updateProject.mutate(
+                {
+                    id: configuration.id,
+                    settings: {
+                        workspaceClasses: {
+                            regular: selectedValue,
+                        },
+                    },
+                },
+                {
+                    onSuccess: () => {
+                        // todo: use optimistic updates when we introduce configuration update hooks
+                        toast({ message: "Workspace size updated" });
+                    },
+                    onError: (e) => {
+                        toast({ message: `Failed updating workspace size: ${e.message}` });
+                    },
+                },
+            );
+        },
+        [configuration.id, selectedValue, toast, updateProject],
+    );
+
+    if (isError) {
+        return <div>Something went wrong</div>;
+    }
+
+    if (isLoading || !classes) {
+        return <div>Loading...</div>;
+    }
+
+    return (
+        <ConfigurationSettingsField className="px-8 py-6">
+            <form onSubmit={setWorkspaceClass}>
+                <div className="mb-4">
+                    <Heading2 asChild>
+                        <h2 className="text-base">Workspace Size Options</h2>
```

Comment:
i think we can squash this and not use asChild since Heading2 renders an h2

Reply:
Oh yea, got a bit confused here yesterday :)